packages/backend-common/src/reading/GerritUrlReader.ts

```diff
@@ -180,6 +183,30 @@ export class GerritUrlReader implements UrlReader {
       throw new NotModifiedError();
     }
 
+    if (
+      this.integration.config.gitilesBaseUrl !== this.integration.config.baseUrl
```

Comment:
Hm. Isn't this going to choose the gitiles route when no gitiles base url is set? The check just looks odd to me. Expected to see something like `if (this.integration.config.gitilesBaseUrl && ...`, maybe I'm missing something :)

Reply:
I'm not going to lie there's still some mystery on this for me too!

From my understanding `gitilesBaseUrl` will defaults to the Gerrit `baseUrl̀` to support cases where Gitiles is not installed (even though I'm not sure if this is a real usecase since most links would be broken)
This is documented at the top of the file:
> To be able to link to Git contents for Gerrit providers in a user friendly
> way we are depending on that there is a Gitiles installation somewhere
> that we can link to. It is perfectly possible to integrate Gerrit with
> Backstage without Gitiles since all API calls goes directly to Gerrit.
> However if Gitiles is configured, readTree will use it to fetch
> an archive instead of cloning the repository.

In this condition, we check that the `gitilesBaseUrl` is different from the `baseUrl` (ie, check that a Gitiles installation is configured) before trying to download the archive from Gitiles.